state-chain/pallets/cf-broadcast/src/tests.rs

```diff
@@ -200,68 +189,43 @@ fn test_abort_after_number_of_attempts_is_equal_to_the_number_of_authorities() {
 	});
 }
 
+// Helper function: make a broadcast to be aborted upon the next failure.
+fn ready_to_abort_broadcast(broadcast_id: BroadcastId) {
+	// Mock when all the possible broadcasts have failed another broadcast, and are
+	// therefore aborted.
+	let mut validators = MockEpochInfo::current_authorities();
+	let nominee = AwaitingBroadcast::<Test, Instance1>::get(broadcast_id).unwrap().nominee;
+	validators.remove(&nominee);
+	FailedBroadcasters::<Test, Instance1>::insert(broadcast_id, validators);
+}
+
 #[test]
-fn broadcasts_aborted_after_all_report_failures_after_retry() {
-	new_test_ext()
-		.execute_with(|| {
-			let (_tx_out_id1, api_call1) = api_call(1);
+fn broadcasts_aborted_after_all_report_failures() {
```

Comment:
we should also add a test for the case where it timesout, that it should also result in aborted

Reply:
We should also add a test that after we have aborted, we can still successfully do `transaction_succeeded` - this case may happen where the transaction was actually broadcast but was stuck in the mempool and so was just witnessed late

Reply:
Added test for when the last broadcaster timed out.

Added test for `transaction_succeeded` on aborted broadcast